### PR TITLE
EREGCSC-1097 Clean up eCFR parser

### PR DESCRIPTION
**Pull request by cgodwin1**

### CODE REVIEW

**Review by theabrad — APPROVED** [review]
looks good to me

File: tools/ecfr-parser/ecfr/ecfr.go [hunk 1]
```diff
@@ -59,42 +58,34 @@ func fetch(ctx context.Context, path *url.URL, opts []FetchOption) (io.Reader, e
...
 	req.Header.Set("User-Agent", "E-regs for "+os.Getenv("NAME"))
-	log.Trace("User Agent is: ", req.Header.Get("User-Agent"))
+	log.Trace("[ecfr] User agent is: ", req.Header.Get("User-Agent"))
 
 	log.Trace("[ecfr] Connecting to ", u.String())
-	reqStart := time.Now()
 	resp, err := client.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("[err] from `client.Do`: %+v, took %+v", err, time.Since(reqStart))
+		return nil, fmt.Errorf("Client failed to connect: %+v", err)
```
Comment (knollfear) [inline tools/ecfr-parser/ecfr/ecfr.go:70]: [ecfr] is removed because this is not trace level logging?

File: tools/ecfr-parser/eregs/eregs.go [hunk 2]
```diff
@@ -86,51 +93,79 @@ func PostPart(ctx context.Context, p *Part) error {
...
 		if err != nil {
-			fmt.Errorf("Received error code %d while posting, unable to extract error message: %+v", err)
+			return fmt.Errorf("Received error code %d while posting to %s, unable to extract error message: %+v", resp.StatusCode, eregsPath.String(), err)
 		}
-		return fmt.Errorf("Received error code %d while posting: %s", resp.StatusCode, eregsError.Exception)
+		return fmt.Errorf("Received error code %d while posting to %s: %s", resp.StatusCode, eregsPath.String(), eregsError.Exception)
 	}
 
-	log.Trace("[eregs] Posted ", length, " bytes for part ", p.Name, " version ", p.Date, " in ", time.Since(start))
+	log.Trace("[eregs] Posted ", length, " bytes to ", eregsPath.String(), " in ", time.Since(start))
```
Comment (knollfear) [inline tools/ecfr-parser/eregs/eregs.go:101]: consider converting to message format 
"[eregs] Posted  %d ..." , length ....

File: tools/ecfr-parser/eregs/eregs.go [hunk 3]
```diff
@@ -86,51 +93,79 @@ func PostPart(ctx context.Context, p *Part) error {
...
+	checkURL, err := url.Parse(BaseURL)
+	if err != nil {
+		return nil, err
+	}
+	q := checkURL.Query()
+	q.Add("json_errors", "true")
+	checkURL.RawQuery = q.Encode()
+	checkURL.Path = path.Join(checkURL.Path, fmt.Sprintf(partURL, title))
+
+	log.Trace("[eregs] Beginning checking of existing parts for title ", title, " at ", checkURL.String())
```
Comment (knollfear) [inline tools/ecfr-parser/eregs/eregs.go:134]: another opportunity for message formatting.

**Review by knollfear — APPROVED** [review]
Good work. If you don't want to convert all of the error messages to message formatting that is fine.

**Pull request merged**